packages/suite-native-atoms/src/Input/Input.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { ReactNode, useRef, useState } from 'react';
+import { TouchableWithoutFeedback, TextInput } from 'react-native';
+
+import { prepareNativeStyle, useNativeStyles } from '@trezor/styles';
+import { Text } from '../Text';
+import { Box } from '../Box';
+
+type InputProps = {
+    value: string;
+    label: string;
+    onChange: (value: string) => void;
+    isDisabled?: boolean;
+    isInvalid?: boolean;
+    hasWarning?: boolean;
+    leftIcon?: ReactNode;
+};
+
+type InputWrapperStyleProps = {
+    hasWarning: boolean;
+    isInvalid: boolean;
+};
+const inputWrapperStyle = prepareNativeStyle<InputWrapperStyleProps>(
+    (utils, { isInvalid, hasWarning }) => ({
+        borderWidth: utils.borders.widths.small,
+        borderColor: utils.colors.gray300,
+        backgroundColor: utils.colors.gray300,
+        borderRadius: utils.borders.radii.small,
+        paddingVertical: utils.spacings.small,
+        paddingHorizontal: 14,
+        height: 58,
+        justifyContent: 'center',
+        extend: [
+            {
+                condition: hasWarning,
```

Comment:
where did you find how this warning state should looks like? I believe that there is nothing like this in Figma. I believe warning will displayed only under input using Hint component (same as error).

Reply:
![image](https://user-images.githubusercontent.com/36101761/173337181-9000cd9e-576b-4428-8fd3-02b57c12c077.png)

Reply:
TODO ask Ondra to elaborate

Reply:
Still waiting for reply but if you're not against it I would just leave this state and then we can refactor later. Or I'll just delete it before rebase.